src/hyp3_isce2/insar_tops_burst.py

```diff
@@ -85,23 +92,89 @@ def topsapp_burst(
+    Specifying the HyP3 interface will add HyP3 specific arguments, such as
+    S3 bucket and prefix. If running locally, use the main interface.
+
+    Args:
+        interface: The interface to use
+    """
+    parser = argparse.ArgumentParser(description=__doc__, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    if interface == 'hyp3':
+        parser.add_argument('--bucket', help='AWS S3 bucket HyP3 for upload the final product(s)')
+        parser.add_argument('--bucket-prefix', default='', help='Add a bucket prefix to product(s)')
+    elif interface == 'main':
+        pass
+    else:
+        raise NotImplementedError(f'Unknown interface: {interface}')
 
-def main():
-    """process_isce2 entrypoint"""
-    parser = ArgumentParser(prog='topsapp_burst', description=__doc__, formatter_class=ArgumentDefaultsHelpFormatter)
-    parser.add_argument('--version', action='version', version=f'%(prog)s {__version__}')
     parser.add_argument('--reference-scene', type=str, required=True)
     parser.add_argument('--secondary-scene', type=str, required=True)
     parser.add_argument('--swath-number', type=int, required=True)
+    parser.add_argument('--polarization', type=str, default='VV')
     parser.add_argument('--reference-burst-number', type=int, required=True)
     parser.add_argument('--secondary-burst-number', type=int, required=True)
-    parser.add_argument('--polarization', type=str, default='VV')
     parser.add_argument('--azimuth-looks', type=int, default=4)
     parser.add_argument('--range-looks', type=int, default=20)
+
+    return parser
+
+
+def hyp3():
+    """ HyP3 entrypoint for the burst workflow
+
+    Uses the HyP3 specific arguments and uploads the product to S3
+    """
+    parser = _get_cli(interface='hyp3')
     args = parser.parse_args()
 
-    topsapp_burst(**args.__dict__)
+    product_dir = run_insar_tops_burst(args)
+
+    if args.bucket:
+        reference_name = (
+            f'{args.reference_scene}_IW{args.swath_number}_{args.polarization}_{args.reference_burst_number}'
+        )
+        secondary_name = (
+            f'{args.secondary_scene}_IW{args.swath_number}_{args.polarization}_{args.secondary_burst_number}'
+        )
+        base_name = f'{reference_name}x{secondary_name}'
+        product_file = make_archive(base_name=base_name, format='zip', base_dir=product_dir)
+        upload_file_to_s3(product_file, args.bucket, args.bucket_prefix)
+        browse_images = product_file.with_suffix('.png')
+        for browse in browse_images:
+            thumbnail = create_thumbnail(browse)
+            upload_file_to_s3(browse, args.bucket, args.bucket_prefix)
+            upload_file_to_s3(thumbnail, args.bucket, args.bucket_prefix)
+
 
+def main():
+    """Main entrypoint for the burst workflow"""
+    parser = _get_cli(interface='main')
+    args = parser.parse_args()
 
-if __name__ == "__main__":
-    main()
+    run_insar_tops_burst(args)
```

Comment:
Dropping the burst specific entry point would me all this could be condensed back into `main`.